### ISSUE TITLE
Mariadb container started with local volume and exited immediately

### ISSUE THREAD

**Issue by yuzhoualauda** [bitnami/bitnami-docker-testlink#59]
**Description**

**Steps to reproduce the issue:**

1. docker run -d --name=mariadb -e MARIADB_ROOT_PASSWORD=root -p 3306:3306 -v /testlink/mariadb_data/:/bitnami bitnami/mariadb:latest
2. docker run -d --name testlink --env-file=./testlink.env -p 8081:80 --link=mariadb:mariadb -v /testlink/testlink_data/:/bitnami bitnami/testlink:latest
3. testlink.env is below:
 - MARIADB_PASSWORD=root
 - TESTLINK_EMAIL=info@example.com
 - TESTLINK_USERNAME=admin
 - TESTLINK_PASSWORD=admin_testlink
4. after above step, testlink working fine, but after sometime, mariadb container exited.

**Describe the results you received:**
Even I docker start mariadb container, it exited immediately.
The following is the log I got from docker logs:

```
Welcome to the Bitnami mariadb container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
Send us your feedback at containers@bitnami.com

nami    INFO  Initializing mariadb
mariadb INFO
mariadb INFO  ########################################################################
mariadb INFO   Installation parameters for mariadb:
mariadb INFO     Persisted data and properties have been restored.
mariadb INFO     Any input specified will not take effect.
mariadb INFO   This installation requires no credentials.
mariadb INFO  ########################################################################
mariadb INFO
nami    INFO  mariadb successfully initialized
INFO  ==> Starting mariadb...
INFO  ==> Starting mysqld_safe...
180109 02:12:37 mysqld_safe Logging to '/opt/bitnami/mariadb/logs/mysqld.log'.
180109 02:12:37 mysqld_safe Starting mysqld daemon with databases from /opt/bitnami/mariadb/data

Welcome to the Bitnami mariadb container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
Send us your feedback at containers@bitnami.com

INFO  ==> Starting mariadb...
INFO  ==> Starting mysqld_safe...
180125 02:14:31 mysqld_safe Logging to '/opt/bitnami/mariadb/logs/mysqld.log'.
180125 02:14:31 mysqld_safe Starting mysqld daemon with databases from /opt/bitnami/mariadb/data

Welcome to the Bitnami mariadb container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
Send us your feedback at containers@bitnami.com

INFO  ==> Starting mariadb...
INFO  ==> Starting mysqld_safe...
Warning: World-writable config file '/opt/bitnami/mariadb/conf/my.cnf' is ignored
Warning: World-writable config file '/opt/bitnami/mariadb/conf/my.cnf' is ignored
180125 02:26:25 mysqld_safe Logging to '/opt/bitnami/mariadb/data/c2afd5a8d6b6.err'.
180125 02:26:25 mysqld_safe Starting mysqld daemon with databases from /opt/bitnami/mariadb/data
```

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version**

- Output of `docker version`:

```
(paste your output here)
```

- Output of `docker info`:

```
Containers: 2
 Running: 1
 Paused: 0
 Stopped: 1
Images: 2
Server Version: 1.12.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 25
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host overlay null
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.0-38-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.303 GiB
Name: ip-172-31-26-68
ID: 3MO2:QFM5:4YH5:L536:URU4:MAWB:JPN6:P7GW:TXFN:IT4J:VIJI:ATLY
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```

- Output of `docker-compose version` (if applicable):

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
AWS EC2

**Comment by juan131** [comment]
I don't see anything weird on the logs you shared. Can you access to the log file `/opt/bitnami/mariadb/data/c2afd5a8d6b6.err`? You might get more information about the reason why MariaDB stopped working on that file.

**Comment by yuzhoualauda** [comment]
Hi @juan131 

Thanks for your quick reply.

The attachment is the error log for your reference.
[error.log](https://github.com/bitnami/bitnami-docker-testlink/files/1663631/error.log)

Thanks

**Comment by juan131** [comment]
I think the problem is that MySQL is unable to write on the Volume you attached for the MySQL data because of a permissions issue. 

Could you please repeat the process without attaching a Volume to the MariaDB container? I mean without adding: `-v /testlink/mariadb_data/:/bitnami`

This is just to confirm that's the issue.

**Comment by yuzhoualauda** [comment]
Hi @juan131 

Without -v, yes, it works.

Then how can i resolve it?

**Comment by juan131** [comment]
Hi @yuzhoualauda 

You need to change the volume ownership configuration before attaching it to your MariaDB container:

```
$ chown -R 1001:1001 /path/to/your/mariadb_data
```

**Comment by yuzhoualauda** [comment]
Hi @juan131 

I execute your cmd, and the problem is still there.

The following is the error log I got from the container.
[error_20180306.log](https://github.com/bitnami/bitnami-docker-testlink/files/1783793/error_20180306.log)

**Comment by silviofdez** [comment]
Hi, @yuzhoualauda,

if you are right the permissions are right, there could also be a corrupted file. MariaDB is able to deal with it, I will comment you how:

Edit my.cnf - add the line: `innodb_force_recovery=2`
(after the `[mysqld]` section)
Restart MariaDB `nami restart mariadb` to let the MariaDB engine repair itself.
Comment the innodb_force_recovery line (add `#` at the beginning)
Restart MariaDB `nami restart mariadb` again and the problem should be gone

**Comment by yuzhoualauda** [comment]
Hi @silviofdez 

I restart the mariadb container and find the my.conf file, but there is no vi cmd and I cannot sudo -s to switch to root user.

Then I add the innodb_force_recovery=2 line into my local file and use -v to add it into mariadb container, and run the container with bash, then run your cmd nami restart mariadb, but meet error `nami    ERROR com.bitnami.mariadb is not fully installed. You cannot execute commands`

Then I run `./app-entrypoint.sh /run.sh` and meet the following error:
`nami    INFO  Initializing mariadb
mariadb INFO
mariadb INFO  ########################################################################
mariadb INFO   Installation parameters for mariadb:
mariadb INFO     Persisted data and properties have been restored.
mariadb INFO     Any input specified will not take effect.
mariadb INFO   This installation requires no credentials.
mariadb INFO  ########################################################################
mariadb INFO
nami    INFO  mariadb successfully initialized
INFO  ==> Starting mariadb...
[WARN  tini (15)] Tini is not running as PID 1 and isn't registered as a child subreaper.
Zombie processes will not be re-parented to Tini, so zombie reaping won't work.
To fix the problem, use the -s option or set the environment variable TINI_SUBREAPER to register Tini as a child subreaper, or run Tini as PID 1.
INFO  ==> Starting mysqld_safe...
Warning: World-writable config file '/opt/bitnami/mariadb/conf/my.cnf' is ignored
Warning: World-writable config file '/opt/bitnami/mariadb/conf/my.cnf' is ignored
180309 04:02:51 mysqld_safe Logging to '/opt/bitnami/mariadb/data/0414b243b53d.err'.
180309 04:02:51 mysqld_safe Starting mysqld daemon with databases from /opt/bitnami/mariadb/data`

And the error log file is: 
[error.log](https://github.com/bitnami/bitnami-docker-testlink/files/1795591/error.log)

**Comment by silviofdez** [comment]
Hi @yuzhoualauda ,

```
World-writable config file '/opt/bitnami/mariadb/conf/my.cnf'
```
this is your problem. In order to solve it, you need to change the permissions of the `my.cnf` on yout **host** system. Correct permissions are `-rw-r--r--`. It could be that you need to change them to allow editing the file, but you should restore them before launch the container.

**Comment by yuzhoualauda** [comment]
Hi @silviofdez 

It is still not working.

Finally I remove both testlink and db containers, and redo all steps from the beginning, not it is working.

Below are the steps:
1. Remove all containers
2. Remove all files which -v used before
3. Create a new local file path and `chown -R 1001:1001 /path/to/your/mariadb_data`
4. Start the mariadb container with -v
5. Start the testlink container with -v

Now it is working fine even I remove both testlink containers and start them again.

Maybe the first time, I did not run cmd `chown -R 1001:1001 /path/to/your/mariadb_data` or I gave 777 permissions to my local file path, then problem happened.

Hope this helpful.

Thanks
